Adding all new demos to release/3.0

FlorianJacta opened the pull request.

jrobinAV committed:
Merge pull request #729 from Avaiga/demo_pollution_dash

:tada: Adds a Realtime Pollution Dashboard Demo

jrobinAV committed:
Merge pull request #732 from Avaiga/doc/add-background-remover-demo

Add Background Remover demo in Doc

FlorianJacta committed:
Merge pull request #740 from Avaiga/doc/add-background-remover-demo

Fix filepath

jrobinAV committed:
Merge pull request #690 from Avaiga/demo-drift-detection

:tada: Adds Drift Detection Demo

AlexandreSajus committed:
Merge pull request #758 from Avaiga/demo_frauddetection

Adds a Credit Card Fraud Detection demo

AlexandreSajus committed:
Merge pull request #756 from Avaiga/chatbot_tutorial

Adds an LLM ChatBot tutorial and demo

AlexandreSajus committed:
Merge pull request #759 from Avaiga/demo_talktotaipy

Adds TalkToTaipy to demos

AlexandreSajus committed:
Merge pull request #764 from Avaiga/broken_link_fraud_demo

Repairs broken fraud demo link

AlexandreSajus committed:
Merge pull request #767 from Avaiga/chatbot_link_fix

Fixes image and link to chatbot demo

AlexandreSajus committed:
Merge pull request #772 from Avaiga/chatbot_demo_div

Fixes Chatbot demo div in knowledge base

jrobinAV reviewed (commented):
knowledge_base/index.md should also be updated.

jrobinAV reviewed (commented):
To help the review, please generate the doc and check that:
- [x] The navigation panel is correct and contains all your new demos
- [x] The `knowledge_base` page is correct and contains all your new demos
- [x] The `knowledge_base/demos` page is correct and contains all your new demos
- [x] The `knowledge_base/tutorials` page is correct and contains all your new tutotials
- [x] The `knowledge_base/tips` page is correct and contains all your new tips
- [x] Each new demo/tip/tuto page is correct, including the title, the sub-titles, the table of contents, and the links.
- [x] Run Grammarly on the generated pages.

FlorianJacta commented:
I have some difficulty in building the doc. This was my plan. My problem is to fetch the release/3.0. I created the PR to see what is missing as files. The PR is currently not complete

FabienLelaquais committed:
Fix links in Page Builder API doc (#778)

- Fix xrefs including from the Page Buider API refman
- Python linter complains
- Basic charts generation missing
- Typography rules

FlorianJacta commented:
The card for 'fraud_detection' appears twice. I am trying to figure out how I did that. It appears twice in index.md of demos. Do you know how I can fix this?

jrobinAV committed:
remove duplicate subtitle

FlorianJacta committed:
Erase dupplicate of fraud detection

FabienLelaquais commented on docs/knowledge_base/demos/background_remover.md:
```diff
@@ -0,0 +1,39 @@
+- **Code Efficiency:** Achieves the background removal task in just a few lines of Python
+code.
+
+This one page demo uses rembg and [OpenCV library](https://opencv.org/) to remove the
```
```suggestion
This one page demo uses `rembg` and [the OpenCV library](https://opencv.org/) to remove the
```
A link to the rembg repo?

FabienLelaquais commented on docs/knowledge_base/demos/background_remover.md:
```diff
@@ -0,0 +1,39 @@
+
+Using Background Remover is straightforward:
+
+1. **Upload Your Image:** Select an image that you want to remove the background; this
```
```suggestion
1. **Upload Your Image:** Select an image where you want to remove the background; this
```

jrobinAV replied:
"In which" instead of where or that.

FabienLelaquais commented on docs/knowledge_base/demos/background_remover.md:
```diff
@@ -0,0 +1,39 @@
+
+1. **Upload Your Image:** Select an image that you want to remove the background; this
+will start the processing.
+2. **More options:** Some parameters influences how the background is removed. These
```
```suggestion
2. **More options:** Some parameters influence how the background is removed. These
```

FabienLelaquais commented on docs/knowledge_base/demos/background_remover.md:
```diff
@@ -0,0 +1,39 @@
+1. **Upload Your Image:** Select an image that you want to remove the background; this
+will start the processing.
+2. **More options:** Some parameters influences how the background is removed. These
+parameters can be changed in the "More options" expandable.
```
```suggestion
parameters can be changed in the "More options" expandable section.
```

FabienLelaquais commented on docs/knowledge_base/demos/background_remover.md:
```diff
@@ -0,0 +1,39 @@
+parameters can be changed in the "More options" expandable.
+3. **Instant Result:** The processed image with the background removed will be displayed
+below.
+4. **Download:** You can download the image through a button.
```
```suggestion
4. **Download:** You can download the image by pressing a button.
```

FabienLelaquais commented on docs/knowledge_base/demos/chatbot.md:
```diff
@@ -0,0 +1,16 @@
+[Get it on GitHub](https://github.com/Avaiga/demo-llm-chat){: .tp-btn .tp-btn--accent target='blank' }
+
+# Understanding the Application
+This application allows the user to chat with GPT-3 by sending 
```
A link to GPT?

FabienLelaquais commented on docs/knowledge_base/demos/drift_detection.md:
```diff
@@ -0,0 +1,48 @@
+is drift.
+
+The dataset used in this app has both numerical (age, body mass index, blood pressure) and categorical 
+(sex) columns. That is why we start by separating these columns first. Then, we run a statistical test 
```
```suggestion
(sex) columns. That is why we start by separating these columns. Then, we run a statistical test 
```

FabienLelaquais commented on docs/knowledge_base/demos/drift_detection.md:
```diff
@@ -0,0 +1,48 @@
+
+1. Select the comparison dataset:
+
+Here, we are selecting data_big, a dataset similar to the dataset of reference but with rows with higher 
```
```suggestion
Here, we are selecting *data_big*, a dataset similar to the dataset of reference but with rows with higher 
```

FabienLelaquais commented on docs/knowledge_base/demos/fraud_detection.md:
```diff
@@ -0,0 +1,22 @@
+[Try it live](https://fraud-detection.taipy.cloud/Transactions?null=){: .tp-btn target='blank' }
+[Get it on GitHub](https://github.com/Avaiga/demo-fraud-detection){: .tp-btn .tp-btn--accent target='blank' }
+
+# Understanding the Application
```
Skip a line after title (for readability and homogeneity)

FabienLelaquais commented on docs/knowledge_base/demos/fraud_detection.md:
```diff
@@ -0,0 +1,22 @@
+![List of Transactions Page](images/fraud_transactions.png){width=100%}
+
+The user can select a transaction to see an explanation of the model's prediction, as well as the client's
+ other transactions.
```
```suggestion
other transactions.
```

FabienLelaquais commented on docs/knowledge_base/demos/fraud_detection.md:
```diff
@@ -0,0 +1,22 @@
+![Prediction Explanation Page](images/fraud_explanation.png){width=100%}
+
+The user can also choose the threshold of the model. The threshold is the output of the model
+ above which a transaction is considered fraudulent. The user can choose the model according
```
```suggestion
above which a transaction is considered fraudulent. The user can choose the model according
```

FabienLelaquais commented on docs/knowledge_base/demos/fraud_detection.md:
```diff
@@ -0,0 +1,22 @@
+
+The user can also choose the threshold of the model. The threshold is the output of the model
+ above which a transaction is considered fraudulent. The user can choose the model according
+  to the displayed confusion matrix and by looking at False Positive and False Negative transactions.
```
```suggestion
to the displayed confusion matrix and by looking at false positive and false negative transactions.
```

FabienLelaquais commented on docs/knowledge_base/demos/chatbot.md:
```diff
@@ -0,0 +1,16 @@
+[Try it live](https://demo-llm-chat.taipy.cloud/){: .tp-btn target='blank' }
+[Get it on GitHub](https://github.com/Avaiga/demo-llm-chat){: .tp-btn .tp-btn--accent target='blank' }
+
+# Understanding the Application
```
Skip a line after title (for readability and homogeneity)

FabienLelaquais commented on docs/knowledge_base/demos/drift_detection.md:
```diff
@@ -0,0 +1,48 @@
+[Try it live](https://drift-detection.taipy.cloud/){: .tp-btn target='blank' }
+[Get it on GitHub](https://github.com/Avaiga/demo-drift-detection){: .tp-btn .tp-btn--accent target='blank' }
+
+# Understanding the Application
```
Skip a line after title (for readability and homogeneity)

FabienLelaquais commented on docs/knowledge_base/demos/pollution_sensors.md:
```diff
@@ -0,0 +1,13 @@
+[Try it live](https://realtime-pollution.taipy.cloud/){: .tp-btn target='blank' }
+[Get it on GitHub](https://github.com/Avaiga/demo-realtime-pollution){: .tp-btn .tp-btn--accent target='blank' }
+
+# Understanding the Application
```
Skip a line after title (for readability and homogeneity)

FabienLelaquais commented on docs/knowledge_base/demos/talk_to_taipy.md:
```diff
@@ -0,0 +1,17 @@
+
+[Try it live](https://talk-to-taipy.taipy.cloud/){: .tp-btn target='blank' }
+
+# Understanding the Application
```
Skip a line after title (for readability and homogeneity)

FabienLelaquais commented on docs/knowledge_base/demos/pollution_sensors.md:
```diff
@@ -0,0 +1,13 @@
+
+# Understanding the Application
+This application shows pollution levels around a factory in real-time. The data is generated on 
+another server and sent to this Taipy app via a WebSocket. Taipy then processes the data and 
```
```suggestion
another server and sent to this Taipy application via a WebSocket. Taipy then processes the data and 
```

FabienLelaquais commented on docs/knowledge_base/tutorials/chatbot/index.md:
````diff
@@ -0,0 +1,275 @@
+current_user_message = ""
+```
+
+- `context` is the initial context for the conversation, the LLM will use this to understand what behaviour is expected from it.
````
```suggestion
- *context* is the initial context for the conversation, the LLM will use this to understand what behavior is expected from it.
```

FabienLelaquais commented on docs/knowledge_base/tutorials/chatbot/index.md:
````diff
@@ -0,0 +1,275 @@
+```
+
+- `context` is the initial context for the conversation, the LLM will use this to understand what behaviour is expected from it.
+- `conversation` is a dictionary that will store the conversation history to be displayed
````
```suggestion
- *conversation* is a dictionary that stores the conversation history to be displayed.
```

FabienLelaquais commented on docs/knowledge_base/tutorials/chatbot/index.md:
```diff
@@ -0,0 +1,275 @@
+
+- `context` is the initial context for the conversation, the LLM will use this to understand what behaviour is expected from it.
+- `conversation` is a dictionary that will store the conversation history to be displayed
+- `current_user_message` is the current message that the user is typing
```
```suggestion
- *current_user_message* is the current message that the user is typing.
```

FabienLelaquais commented on docs/knowledge_base/tutorials/chatbot/index.md:
````diff
@@ -0,0 +1,275 @@
+client = openai.Client(api_key="ENTER_YOUR_API_KEY_HERE")
+```
+
+Create a function that takes as input a string `prompt` which
````
```suggestion
Create a function that takes as input the string stored in *prompt* which
```

FabienLelaquais commented on docs/knowledge_base/tutorials/chatbot/index.md:
```diff
@@ -0,0 +1,275 @@
+    Send a prompt to the GPT-3 API and return the response.
+
+    Args:
+        - state: The current state of the app.
```
```suggestion
        - state: The current state.
```

FabienLelaquais commented on docs/knowledge_base/tutorials/chatbot/index.md:
```diff
@@ -0,0 +1,275 @@
+# Step 5: Create a function to add the new messages to the conversation
+
+Create a function that gets triggered when the user sends a
+message. This function will add the user's message to the context,
```
```suggestion
message. This function adds the user's message to the context,
```

FabienLelaquais commented on docs/knowledge_base/tutorials/chatbot/index.md:
```diff
@@ -0,0 +1,275 @@
+
+Create a function that gets triggered when the user sends a
+message. This function will add the user's message to the context,
+send it to the API, get the response, add the response to the
```
```suggestion
sends it to the API, gets the response, and adds the response to the
```

FabienLelaquais commented on docs/knowledge_base/tutorials/chatbot/index.md:
```diff
@@ -0,0 +1,275 @@
+    Send the user's message to the API and update the conversation.
+
+    Args:
+        - state: The current state of the app.
```
```suggestion
        - state: The current state.
```

FabienLelaquais commented on docs/knowledge_base/tutorials/chatbot/index.md:
```diff
@@ -0,0 +1,275 @@
+# Step 6: Create the User Interface
+
+In Taipy, one way to define pages is to use Markdown strings. Here we use a
+[table](../../../manuals/gui/viselements/table.md_template) to display the
```
```suggestion
[table](../../../manuals/gui/viselements/table.md) to display the
```

FabienLelaquais commented on docs/knowledge_base/tutorials/chatbot/index.md:
```diff
@@ -0,0 +1,275 @@
+
+In Taipy, one way to define pages is to use Markdown strings. Here we use a
+[table](../../../manuals/gui/viselements/table.md_template) to display the
+ `conversation` dictionary and an
```
```suggestion
*conversation* dictionary and an
```

FabienLelaquais commented on docs/knowledge_base/tutorials/chatbot/index.md:
```diff
@@ -0,0 +1,275 @@
+In Taipy, one way to define pages is to use Markdown strings. Here we use a
+[table](../../../manuals/gui/viselements/table.md_template) to display the
+ `conversation` dictionary and an
+ [input](../../../manuals/gui/viselements/input) so that the
```
```suggestion
 [input](../../../manuals/gui/viselements/input.md) so that the
```

FabienLelaquais commented on docs/knowledge_base/tutorials/chatbot/index.md:
```diff
@@ -0,0 +1,275 @@
+ `conversation` dictionary and an
+ [input](../../../manuals/gui/viselements/input) so that the
+ user can type their message. When the user presses enter,
+ the `send_message` function is triggered.
```
```suggestion
 the *send_message()* function is triggered.
```

FabienLelaquais commented on docs/knowledge_base/tutorials/chatbot/index.md:
````diff
@@ -0,0 +1,275 @@
+
+```python
+if __name__ == "__main__":
+    Gui(page).run(dark_mode=True, title="Taipy Chat")
````
```suggestion
    Gui(page).run(title="Taipy Chat")
```

FabienLelaquais commented on docs/knowledge_base/tutorials/chatbot/index.md:
```diff
@@ -0,0 +1,275 @@
+
+# Step 8: Styling
+
+The app's style is Taipy's default stylekit. We are going to
```
```suggestion
The application's style is Taipy's default Stylekit. We are going to
```

FabienLelaquais commented on docs/knowledge_base/tutorials/chatbot/index.md:
````diff
@@ -0,0 +1,275 @@
+}
+```
+
+We now need to tell Taipy to apply these styles to the rows in
````
```suggestion
We now need to get Taipy to apply these styles to the rows in
```

FabienLelaquais commented on docs/knowledge_base/tutorials/chatbot/index.md:
````diff
@@ -0,0 +1,275 @@
+```
+
+We now need to tell Taipy to apply these styles to the rows in
+the table. We'll first create a function that will return the
````
```suggestion
the table. We'll first create a function that returns the
```

FabienLelaquais commented on docs/knowledge_base/tutorials/chatbot/index.md:
```diff
@@ -0,0 +1,275 @@
+    Apply a style to the conversation table depending on the message's author.
+
+    Args:
+        - state: The current state of the app.
```
```suggestion
        - state: The current state.
```

FabienLelaquais commented on docs/knowledge_base/tutorials/chatbot/index.md:
```diff
@@ -0,0 +1,275 @@
+
+# Step 9: More features
+
+I have added notifications, a sidebar with a button to clear the conversation
```
"I"??

FabienLelaquais commented on docs/knowledge_base/tutorials/chatbot/index.md:
```diff
@@ -0,0 +1,275 @@
+# Step 9: More features
+
+I have added notifications, a sidebar with a button to clear the conversation
+and a history of previous conversations. I won't go into the
```
"I"??

FabienLelaquais commented on docs/knowledge_base/tutorials/chatbot/index.md:
```diff
@@ -0,0 +1,275 @@
+
+# Step 10: Deploying the app to Taipy Cloud
+
+We are now going to deploy the app to Taipy Cloud so it is
```
```suggestion
We are now going to deploy the application to Taipy Cloud so it is
```

FabienLelaquais commented on docs/knowledge_base/tutorials/chatbot/index.md:
```diff
@@ -0,0 +1,275 @@
+Now, instead of having our API key in the code, the app will look
+for it in the environment variables.
+
+We can now deploy the app to Taipy Cloud:
```
Skip a line, or you won't get a proper bullet list

FabienLelaquais reviewed (changes requested):
A few blocking minor things...

FabienLelaquais commented:
> To help the review, please generate the doc and check that:
> 
> * [ ]  The navigation panel is correct and contains all your new demos
> * [ ]  The `knowledge_base` page is correct and contains all your new demos
> * [ ]  The `knowledge_base/demos` page is correct and contains all your new demos
> * [ ]  The `knowledge_base/tutorials` page is correct and contains all your new tutos
> * [ ]  The `knowledge_base/tips` page is correct and contains all your new tips
> * [ ]  Each new demo/tip/tuto page is correct, including the title, the sub-titles, the table of content, the links.

I would add... run Grammarly on the generated pages.

FlorianJacta committed:
Merge pull request #782 from Avaiga/doc/fix-demos

Rephrase demos

FlorianJacta committed:
Fix typo

FlorianJacta committed:
Rephrase chatbot tutorial

FabienLelaquais reviewed (approved):
👌
